src/tools/cephfs/cephfs-shell

```diff
@@ -0,0 +1,830 @@
+    locate_parser.add_argument('name', help = 'name', type = str)
+    locate_parser.add_argument('-c', '--count', action = 'store_true', help = 'Count list of items located.')
+    locate_parser.add_argument('-i', '--ignorecase', action = 'store_true', help = 'Ignore case')
+
+    @with_argparser(locate_parser)
+    def do_locate(self, args):
+        """
+        Find a file within the File System
+        """
+        if args.name.count('*') == 1:
+            if args.name[0] == '*':
+                args.name += '/'
+            elif args.name[-1] == '*':
+                args.name = '/'+ args.name
+        args.name = args.name.replace('*', '')
+        if args.ignorecase:
+            locations = [i for i in sorted(set(dirwalk(cephfs.getcwd().decode('utf-8')))) if args.name.lower() in i.lower()]
+        else:
+            locations = [i for i in sorted(set(dirwalk(cephfs.getcwd().decode('utf-8')))) if args.name in i]
+        if args.count:
+            print(len(locations))
+        else:
+            print('\n'.join(locations))
+
+    def complete_du(self, text, line, begidx, endidx):
+        """ 
+        auto complete of file name.
+        """    
+        return self.complete_filenames(text, line, begidx, endidx)
+
+    du_parser = argparse.ArgumentParser(description = 'Disk Usage of a Directory')
+    du_parser.add_argument('dirs', type = str, help = 'Name of the directory.', nargs = '?', default = '')
+    du_parser.add_argument('-r', action = 'store_true', help = 'Recursive Disk usage of all directories.')
+    
+    @with_argparser(du_parser)
+    def do_du(self, args):
+        """
+        Disk Usage of a Directory
+        """
+        if args.dirs == '':
+            args.dirs = cephfs.getcwd().decode('utf-8') 
+        for dir_ in args.dirs:
+            if args.r :   
+                for i in reversed(sorted(set(dirwalk(dir_)))):
+                    try:
+                        print('{:10s} {}'.format(humansize(int(cephfs.getxattr(to_bytes(i), 'ceph.dir.rbytes').decode('utf-8'))), '.' + re.sub('\/+', '/', i)))
+                    except:
+                        continue
+            else :
+                 print('{:10s} {}'.format(humansize(int(cephfs.getxattr(to_bytes(dir_), 'ceph.dir.rbytes').decode('utf-8'))), '.' + re.sub('\/+', '/', dir_)))
+      
+    def do_help(self, line):
+        """
+        Get details about a command.
+            Usage: help <cmd> - for a specific command
+                   help all - for all the commands
+        """
+        if line == 'all':
+            for k in dir(self):
+                if k.startswith('do_'):
+                    print('-'*80)
+                    super().do_help(k[3:])
+            return
+        parser = self.create_argparser(line)
+        if parser:
+            parser.print_help()
+        else:
+            super().do_help(line)
+
+ 
+if __name__ == '__main__':
+    setup_cephfs()
```

Comment:
Add argument parsing here. At the very least we'll want a `-c` option which lets' us set the config file here: https://github.com/ceph/ceph/pull/23158/files#diff-9226892dd49173c4687a9953904ccaa8R24